pallets/runtime/tests/src/identity_test.rs

```diff
@@ -322,6 +338,63 @@ fn only_primary_key_can_add_secondary_key_permissions_with_externalities() {
     ));
 }
 
+#[test]
+fn add_permissions_to_multiple_tokens() {
+    ExtBuilder::default()
+        .monied(true)
+        .build()
+        .execute_with(&do_add_permissions_to_multiple_tokens);
+}
+fn do_add_permissions_to_multiple_tokens() {
+    let bob_key = AccountKeyring::Bob.public();
+    let bob_signer = Signatory::Account(bob_key);
+    let alice = User::new(AccountKeyring::Alice);
+
+    // Add bob with default permissions
+    add_secondary_key(alice.did, bob_signer);
+
+    // Create some tokens
+    let max_tokens = 30;
+    let tokens: Vec<Ticker> = (0..max_tokens)
+        .map(|i| {
+            let name = format!("TOKEN_{}", i);
+            let (ticker, _) = create_new_token(name.as_bytes(), alice);
+            ticker
+        })
+        .collect();
+
+    let set_bob_asset_permissions = |num_token| {
+        assert_ok!(Identity::set_permission_to_signer(
+            alice.origin(),
+            bob_signer,
+            Permissions {
+                asset: AssetPermissions::elems(tokens[0..num_token].into_iter().map(|t| t.clone())),
+                ..Default::default()
+            },
+        ));
+    };
+
+    // add one-by-one
+    for num in 0..max_tokens {
+        set_bob_asset_permissions(num);
+    }
+
+    // remove all permissions
+    set_bob_asset_permissions(0);
+
+    // bulk add, reverse order
+    assert_ok!(Identity::set_permission_to_signer(
+        alice.origin(),
+        bob_signer,
+        Permissions {
+            asset: AssetPermissions::elems(
+                tokens[0..max_tokens].into_iter().rev().map(|t| t.clone())
```

Comment:
Hmm; assert something about the order in which the permissions end up in when converting the `BTreeSet` into a vector?

Reply:
I added the `rev` just to see if it could cause the `Vec` -> `BTreeSet` issue.  But the tests don't do encode/decode and they wouldn't be using the `Vec`-hack.  I just left the `rev` in.